### PR TITLE
error if file with same id exists

### PR DESCRIPTION
**Pull request by blankdots**
this is to prevent message requeuing so that the finalize gets blocked

- we check the id exists before we try to do anything with it and try to mark it as ready.
- add file for integration test (we skip for sftp header)
- update sda-db to v2.1.9

### CODE REVIEW

**Comment by blankdots** [comment]
related to https://github.com/neicnordic/sda-db/issues/83

**Comment by pontus** [comment]
Looks to me like the logic is inverted (i.e. as you'd expect in C), am I just conused?

I know this is just a stop gap, would be glad if we got a test as well.

**Comment by blankdots** [comment]
> Looks to me like the logic is inverted (i.e. as you'd expect in C), am I just conused?

i had an extra file, does this look better ?

will try to see a test

File: cmd/finalize/finalize.go [hunk 1]
```diff
@@ -155,9 +156,13 @@ func main() {
 					message.AccessionID,
 					message.DecryptedChecksums,
 					err)
-
+				requeueMsg := true
+				if !strings.Contains(err.Error(), "duplicate key value violates unique constraint") {
+					requeueMsg = false
+				}
```
Comment (pontus) [inline cmd/finalize/finalize.go:162]: Ask not to requeue if the error isn't duplicate key - keep requeueing if it is?

Comment (blankdots) [reply cmd/finalize/finalize.go:162]: yes

Comment (blankdots) [reply cmd/finalize/finalize.go]: :facepalm: did not see the `!`

**Comment by blankdots** [comment]
marked as draft till tests are done

File: internal/database/db.go [hunk 2]
```diff
@@ -362,6 +362,45 @@ func (dbs *SQLdb) setArchived(file FileInfo, id int64) error {
...
+}
+
+// checkAccessionIdExists validates if an accessionID exists in the db
+func (dbs *SQLdb) checkAccessionIDExists(accessionID, filepath, checksum string) (bool, error) {
+	dbs.checkAndReconnectIfNeeded()
+
+	db := dbs.DB
+
+	const checkIDExist = "SELECT COUNT(stable_id) FROM local_ega.files WHERE stable_id = $1 " +
+		"and inbox_path = $2 and decrypted_file_checksum = $3 and status = 'READY';"
```
Comment (norling) [inline internal/database/db.go:390]: Explicitly checking for path, checksum and staus might be a mistake here, as there could exist files with different value of these fields, but the same stable_id, and they'll still cause a unique key collision. Something like `SELECT COUNT(*) FROM sda.files WHERE stable_id = $1` is likely sufficient and safer. :)

**Review by norling — APPROVED** [review]

**Review by jbygdell — APPROVED** [review]

**Pull request merged**